### DISCUSSION TITLE
Add scoped database driver similar scoped filesystem driver

### DISCUSSION THREAD

**Discussion by henriquecm** [laravel/framework#44721] [Ideas]
Sometimes I need to connect more than one database on the project. 
But most settings use the same value, usually only database, user and password change.
It would be nice to have a scoped driver for database that works like the scoped driver that was introduced in 9.30 for file systems.

**Comment by Sophist-UK** [comment]
I guess the equivalent to the scoped filesystem driver would be to reuse an existing database driver but with a different prefix. Is that what you are suggesting? Because if so, then I have a use-case for such functionality also.

**Reply by henriquecm** [reply]
Not exactly.
My suggestion is that when you make a scoped driver all the setting would be "copied" to the new driver, and only the ones you define would be overwritten.
This would be useful for when you have more than one database, but all the settings, except login, are the same.

Example: 
```php

        'mysql' => [
            'driver' => 'mysql',
            'url' => env('DATABASE_URL'),
            'host' => env('DB_HOST', '127.0.0.1'),
            'port' => env('DB_PORT', '3306'),
            'database' => env('DB_DATABASE', 'forge'),
            'username' => env('DB_USERNAME', 'forge'),
            'password' => env('DB_PASSWORD', ''),
            'unix_socket' => env('DB_SOCKET', ''),
            'charset' => 'utf8mb4',
            'collation' => 'utf8mb4_unicode_ci',
            'prefix' => '',
            'prefix_indexes' => true,
            'strict' => true,
            'engine' => null,
            'options' => extension_loaded('pdo_mysql') ? array_filter([
                PDO::MYSQL_ATTR_SSL_CA => env('MYSQL_ATTR_SSL_CA'),
            ]) : [],
        ],

        'mysql_scoped' => [
            'driver' => 'scoped',
            'connection' => 'mysql',
            'database' => env('OTHER_DB_DATABASE', 'forge'),
            'username' => env('OTHER_DB_USERNAME', 'forge'),
            'password' => env('OTHER_DB_PASSWORD', ''),
        ],
```
